templates/dotnet/src/Appwrite/Extensions/Extensions.cs.twig

```diff
@@ -0,0 +1,627 @@
+﻿using Newtonsoft.Json;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace {{ spec.title | caseUcfirst }}.Extensions
+{
+    public static class Extensions
+    {
+        public static string ToJson(this Dictionary<string, object?> dict)
+        {
+            return JsonConvert.SerializeObject(dict, Client.SerializerSettings);
+        }
+
+        public static string ToQueryString(this Dictionary<string, object?> parameters)
+        {
+            var query = new List<string>();
+
+            foreach (var kvp in parameters)
+            {
+                switch (kvp.Value)
+                {
+                    case null:
+                        continue;
+                    case IList list:
+                        foreach (var item in list)
+                        {
+                            query.Add($"{kvp.Key}[]={item}");
+                        }
+                        break;
+                    default:
+                        query.Add($"{kvp.Key}={kvp.Value.ToString()}");
+                        break;
+                }
+            }
+
+            return Uri.EscapeUriString(string.Join("&", query));
+        }
+
+        private static IDictionary<string, string> _mappings = new Dictionary<string, string>(StringComparer.InvariantCultureIgnoreCase) {
+
+                #region Mime Types
+                {".323", "text/h323"},
+                {".3g2", "video/3gpp2"},
```

Comment:
Couldn't find a clean way to get mime type from a file without doing this or adding a dependency, the standard library way is missing support for a lot of file types and isn't configurable.

It is ultimately used [here](https://github.com/appwrite/sdk-generator/blob/6935c149b92aef0fa4a64f1c8012a0361e2a2acc/templates/dotnet/src/Appwrite/Models/InputFile.cs.twig#L18) to get a mime type from a file path for uploads

Reply:
This would mean file type will be based on file extension will that always be a good idea, how does the standard library does it for supported types? If standard library uses different method than file extension (reading the file head or something) May be we use the standard library for supported types and only when standard library fails, use this method?

Reply:
Not the best way, but this is the same way the standard library is doing it too (same for at least python, ruby, swift)